core/src/main/scala/io/projectglow/plink/PlinkFileFormat.scala

```diff
@@ -0,0 +1,300 @@
+        case (variant, gtBlock) =>
+          rowConverter.convertRow(variant, sampleIds, gtBlock)
+      }
+    }
+  }
+}
+
+object PlinkFileFormat extends HlsUsageLogging {
+
+  import io.projectglow.common.VariantSchemas._
+
+  // Delimiter options
+  val FAM_DELIMITER_KEY = "famDelimiter"
+  val BIM_DELIMITER_KEY = "bimDelimiter"
+  val DEFAULT_FAM_DELIMITER_VALUE = " "
+  val DEFAULT_BIM_DELIMITER_VALUE = "\t"
+  val CSV_DELIMITER_KEY = "delimiter"
+
+  // Path options
+  val FAM_PATH_KEY = "fam"
+  val BIM_PATH_KEY = "bim"
+
+  // Sample ID options
+  val MERGE_FID_IID = "mergeFidIid"
+
+  val FAM_FILE_EXTENSION = ".fam"
+  val BIM_FILE_EXTENSION = ".bim"
+
+  val BLOCKS_PER_BYTE = 4
+  val MAGIC_BYTES: Seq[Byte] = Seq(0x6c, 0x1b, 0x01).map(_.toByte)
+  val NUM_MAGIC_BYTES: Int = MAGIC_BYTES.size
+
+  /* If a custom FAM or BIM path is not specified, use the BED's prefix as the default prefix  */
+  def getPrefixPath(options: Map[String, String]): String = {
+    val bedPath =
+      options.getOrElse("path", throw new IllegalArgumentException("Must provide path."))
+    bedPath.dropRight(4)
+  }
+
+  /* Parses a FAM file to get the sample IDs */
+  def getSampleIds(options: Map[String, String], hadoopConf: Configuration): Array[String] = {
+    val famPathStr = options.get(FAM_PATH_KEY) match {
+      case Some(s) => s
+      case None => getPrefixPath(options) + FAM_FILE_EXTENSION
+    }
+    logger.info(s"Using FAM file $famPathStr")
+
+    val famDelimiterOption = options.getOrElse(FAM_DELIMITER_KEY, DEFAULT_FAM_DELIMITER_VALUE)
+    val parsedOptions =
+      new CSVOptions(
+        Map(CSV_DELIMITER_KEY -> famDelimiterOption),
+        SQLConf.get.csvColumnPruning,
+        SQLConf.get.sessionLocalTimeZone)
+
+    val mergeFidIid = try {
+      options.getOrElse(MERGE_FID_IID, "true").toBoolean
+    } catch {
+      case _: IllegalArgumentException =>
+        throw new IllegalArgumentException(
+          s"Value for $MERGE_FID_IID must be [true, false]. Provided: ${options(MERGE_FID_IID)}")
+    }
+
+    val famPath = new Path(famPathStr)
+    val hadoopFs = famPath.getFileSystem(hadoopConf)
+    val stream = hadoopFs.open(famPath)
+    val lines = IOUtils.lineIterator(stream, "UTF-8").asScala
+    val filteredLines = CSVUtils.filterCommentAndEmpty(lines, parsedOptions)
+    val parser = new CsvParser(parsedOptions.asParserSettings)
+
+    try {
+      filteredLines.map { l =>
+        val sampleLine = parser.parseRecord(l)
+        require(
+          sampleLine.getValues.length == 6,
+          s"Failed while parsing FAM file $famPath: does not have 6 columns delimited by '$famDelimiterOption'")
+        val individualId = sampleLine.getString(1)
+        if (mergeFidIid) {
+          val familyId = sampleLine.getString(0)
+          s"${familyId}_$individualId"
+        } else {
+          individualId
+        }
+      }.toArray
+    } finally {
+      stream.close()
+    }
+  }
+
+  /* Parses a BIM file to get the variants */
```

Comment:
Can you document more clearly what the output schema is for the rows returned here?